src/OpenApi/test/Services/OpenApiSchemaService/OpenApiComponentService.ParameterSchemas.cs

```diff
@@ -154,6 +155,12 @@ public async Task GetOpenApiParameters_HandlesRouteParameterWithRouteConstraint(
         [(TaskStatus status = TaskStatus.Canceled) => { }, (IOpenApiAny defaultValue) => Assert.Equal(6, ((OpenApiInteger)defaultValue).Value)],
         // Default value for enums is serialized as string when a converter is registered.
         [(Status status = Status.Pending) => { }, (IOpenApiAny defaultValue) => Assert.Equal("Pending", ((OpenApiString)defaultValue).Value)],
+        [([DefaultValue(2)] int id) => { }, (IOpenApiAny defaultValue) => Assert.Equal(2, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue(3f)] float id) => { }, (IOpenApiAny defaultValue) => Assert.Equal(3, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue("test")] string id) => { }, (IOpenApiAny defaultValue) => Assert.Equal("test", ((OpenApiString)defaultValue).Value)],
+        [([DefaultValue(true)] bool id) => { }, (IOpenApiAny defaultValue) => Assert.True(((OpenApiBoolean)defaultValue).Value)],
+        [([DefaultValue(TaskStatus.Canceled)] TaskStatus status) => { }, (IOpenApiAny defaultValue) => Assert.Equal(6, ((OpenApiInteger)defaultValue).Value)],
+        [([DefaultValue(Status.Pending)] Status status) => { }, (IOpenApiAny defaultValue) => Assert.Equal("Pending", ((OpenApiString)defaultValue).Value)],
```

Comment:
Based on some [recent bugs](https://github.com/domaindrivendev/Swashbuckle.AspNetCore/issues?q=is%3Aissue+defaultvalue+is%3Aopen+created%3A%3E2024-05-01+) we've had in Swashbuckle, you might want to add some test cases for:

- `[DefaultValue(null)] int? id`
- `[DefaultValue(2)] int? id`
- `[DefaultValue(null)] string? id`
- `[DefaultValue("foo")] string? id`
- `[DefaultValue(null)] TaskStatus? status`
- `[DefaultValue(TaskStatus.Canceled)] TaskStatus? status`

Reply:
Great list! Thank you for sharing. I'll add these. I suspect that `[DefaultValue(null)]` will be interesting to sus out.